HPCC-26916 Re-implement gitfile to use gitlib2 instead of child processes

ghalliday opened the pull request:
Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

ghalliday committed:
HPCC-26916 Re-implement gitfile to use gitlib2 instead of child processes

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

ghalliday commented:
Initial push to check how it compiles on windows...

HPCCSmoketest commented:
Process of PR-15626, label: issue26916 starts now.
The reason of this test is: Forced to re-test.
Commit ID: de40bd58538982bf99c073bece0cb6c4b6cf7527
Estimated completion time is 0.75 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 Host: ip-10-20-0-136.ca-central-1.compute.internal
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Git: 2.9.5,  CMake: 3.18.0,  cUrl: 7.67.0,  node.js: v16.13.0,  npm: 8.1.0

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 Host: ip-10-20-0-136.ca-central-1.compute.internal
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Git: 2.9.5,  CMake: 3.18.0,  cUrl: 7.67.0,  node.js: v16.13.0,  npm: 8.1.0
Sha: de40bd58538982bf99c073bece0cb6c4b6cf7527
Containerized:False
Build: success
Milestone:Install hpccsystems-platform-community_8.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |222 |222 |0 |0 |0 |63 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |9 |9 |0 |21 sec  (00:00:21) |
|setup (thor) |9 |9 |0 |38 sec  (00:00:38) |
|setup (roxie) |19 |19 |0 |17 sec  (00:00:17) |
|test (hthor) |983 |983 |0 |807 sec  (00:13:27) |
|test (thor) |893 |893 |0 |977 sec  (00:16:17) |
|test (roxie) |1052 |1049 |3 |1192 sec  (00:19:52) |


[Errors:](https://gist.github.com/04de7a7b074525c50f2ff6472f2f8f2a)
- test
  - roxie
    - 680\. [schedule2](https://gist.github.com/HPCCSmoketest/04de7a7b074525c50f2ff6472f2f8f2a#file-schedule2-w20220104-173359-1-log), 1012\. [workflow_9c (parallel=false)](https://gist.github.com/HPCCSmoketest/04de7a7b074525c50f2ff6472f2f8f2a#file-workflow_9c-parallel-false-w20220104-174010-5-log), 1013\. [workflow_9c (parallel=true)](https://gist.github.com/HPCCSmoketest/04de7a7b074525c50f2ff6472f2f8f2a#file-workflow_9c-parallel-true-w20220104-174011-log), 

HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   20 sec (00:00:20) |  436 sec (00:07:16) |  105 sec (00:01:45) |   22 sec (00:00:22) |   15 sec (00:00:15) | 1446 sec (00:24:06) |   17 sec (00:00:17) | 2061 sec (00:34:21) |

HPCCSmoketest commented:
Process of PR-15626, label: issue26916 starts now.
The reason of this test is: Forced to re-test.
Commit ID: de40bd58538982bf99c073bece0cb6c4b6cf7527
Estimated completion time is 0.75 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 Host: ip-10-20-0-249.ca-central-1.compute.internal
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Git: 2.9.5,  CMake: 3.18.0,  cUrl: 7.67.0,  node.js: v16.13.0,  npm: 8.1.0

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 Host: ip-10-20-0-249.ca-central-1.compute.internal
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Git: 2.9.5,  CMake: 3.18.0,  cUrl: 7.67.0,  node.js: v16.13.0,  npm: 8.1.0
Sha: de40bd58538982bf99c073bece0cb6c4b6cf7527
Containerized:False
Build: success
Milestone:Install hpccsystems-platform-community_8.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |222 |222 |0 |0 |0 |64 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |9 |9 |0 |21 sec  (00:00:21) |
|setup (thor) |9 |9 |0 |38 sec  (00:00:38) |
|setup (roxie) |19 |19 |0 |18 sec  (00:00:18) |
|test (hthor) |983 |983 |0 |823 sec  (00:13:43) |
|test (thor) |893 |893 |0 |993 sec  (00:16:33) |
|test (roxie) |1052 |1052 |0 |873 sec  (00:14:33) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   19 sec (00:00:19) |  430 sec (00:07:10) |  103 sec (00:01:43) |   19 sec (00:00:19) |   16 sec (00:00:16) | 1247 sec (00:20:47) |   18 sec (00:00:18) | 1852 sec (00:30:52) |

richardkchapman commented on common/remote/hooks/git/libgit2.cmake:
```diff
@@ -0,0 +1,48 @@
+#####################################################
+
+
+if(1)
```
Why?

richardkchapman commented on common/remote/hooks/git/gitfile.cpp:
```diff
@@ -410,10 +413,329 @@ class GitRepositoryDirectoryIterator : implements IDirectoryIterator, public CIn
+
+//--------------------------------------------------------------------------------------------------------------------
+
+static constexpr bool useLibGit = true;
```
I assume this will be removed (and the old code removed) once new is proved to work?

richardkchapman commented on common/remote/hooks/git/gitfile.cpp:
```diff
@@ -410,10 +413,329 @@ class GitRepositoryDirectoryIterator : implements IDirectoryIterator, public CIn
+    }
+    virtual bool getModifiedTime(CDateTime &ret)
+    {
+        UNIMPLEMENTED;
```
Will it be?

ghalliday replied:
It wasn't implemented in the previous code - so probably not unless it is needed.

richardkchapman reviewed (commented):
Some initial comments. Looks like a very good start

richardkchapman commented on common/remote/hooks/git/gitfile.cpp:
```diff
@@ -410,10 +413,329 @@ class GitRepositoryDirectoryIterator : implements IDirectoryIterator, public CIn
+// New implementation using libgit2
+//--------------------------------------------------------------------------------------------------------------------
+
+#define GIT_CHECK(x) check(x, #x)
```
style: This appears to be used inconsistently below

richardkchapman commented on common/remote/hooks/git/gitfile.cpp:
```diff
@@ -410,10 +413,329 @@ class GitRepositoryDirectoryIterator : implements IDirectoryIterator, public CIn
+class NewGitRepositoryFileIO : implements CSimpleInterfaceOf<IFileIO>
+{
+public:
+    NewGitRepositoryFileIO(const char * gitDirectory, const char * revision, const char * relFileName, GitCommitTree * commitTree, const git_oid * oid)
```
First three parameters appear to be unused

ghalliday replied:
removed

richardkchapman commented on common/remote/hooks/git/gitfile.cpp:
```diff
@@ -410,10 +413,329 @@ class GitRepositoryDirectoryIterator : implements IDirectoryIterator, public CIn
+    MemoryBuffer buf;
+};
+
+class NewGitRepositoryFile : public GitRepositoryFile
```
There is code around line 274 that creates a GitRepositoryFile object directly that may need chaging

ghalliday replied:
well spotted, I will change

richardkchapman reviewed (changes requested):
Build errors on Windows and OSX need addressing

HPCCSmoketest commented:
Process of PR-15626, label: issue26916 starts now.
The reason of this test is: Forced to re-test.
Commit ID: c45f1c604327c2be9335ba587fbb7feb015713fc
Estimated completion time is 0.75 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 Host: ip-10-20-0-24.ca-central-1.compute.internal
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Git: 1.8.3.1,  CMake: 3.18.0,  cUrl: 7.67.0,  node.js: v16.13.0,  npm: 8.1.0

HPCCSmoketest commented:
Process of PR-15626, label: issue26916 starts now.
The reason of this test is: Forced to re-test.
Commit ID: c45f1c604327c2be9335ba587fbb7feb015713fc
Estimated completion time is 0.75 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 Host: ip-10-20-0-118.ca-central-1.compute.internal
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Git: 1.8.3.1,  CMake: 3.18.0,  cUrl: 7.67.0,  node.js: v16.13.0,  npm: 8.1.0

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 Host: ip-10-20-0-118.ca-central-1.compute.internal
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Git: 1.8.3.1,  CMake: 3.18.0,  cUrl: 7.67.0,  node.js: v16.13.0,  npm: 8.1.0
Sha: c45f1c604327c2be9335ba587fbb7feb015713fc
Containerized:False
Number of Error(s): 4
collect2: error: ld returned 1 exit status
make[2]: *** [common/remote/hooks/git/CMakeFiles/gitfile.dir/build.make:110: RelWithDebInfo/libs/libgitfile.so] Error 1
make[1]: *** [CMakeFiles/Makefile2:10416: common/remote/hooks/git/CMakeFiles/gitfile.dir/all] Error 2
make: *** [Makefile:182: all] Error 2
Get ECLWatch build result
ReportTimes.
Finish at: 2022-01-06_17-51-37
time logs started
Prep time   :    19 sec (00:00:19)
Build time  :   147 sec (00:02:27)
Package time:     0 sec (00:00:00)
Install time:     0 sec (00:00:00)
Start time  :     0 sec (00:00:00)
Test time   :     0 sec (00:00:00)
Stop time   :     0 sec (00:00:00)
Summary     :   166 sec (00:02:46)
time logs end

[Build error:](https://gist.github.com/32ac39a07c69e5736cd49cb03d93827a)
Build log [Build log](https://gist.github.com/HPCCSmoketest/32ac39a07c69e5736cd49cb03d93827a#file-result-22-01-06-17-48-32-log)

HPCCSmoketest commented:
Process of PR-15626, label: issue26916 starts now.
The reason of this test is: Forced to re-test.
Commit ID: 4dedc3beed6724f9ffd58dcec9af101a9f77e191
Estimated completion time is 0.75 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 Host: ip-10-20-0-189.ca-central-1.compute.internal
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Git: 2.9.5,  CMake: 3.22.1,  cUrl: 7.67.0,  node.js: v16.13.1,  npm: 8.1.2

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 Host: ip-10-20-0-189.ca-central-1.compute.internal
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Git: 2.9.5,  CMake: 3.22.1,  cUrl: 7.67.0,  node.js: v16.13.1,  npm: 8.1.2
Sha: 4dedc3beed6724f9ffd58dcec9af101a9f77e191
Containerized:False
Build: success
Milestone:Install hpccsystems-platform-community_8.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |222 |222 |0 |0 |0 |65 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |9 |9 |0 |21 sec  (00:00:21) |
|setup (thor) |9 |9 |0 |38 sec  (00:00:38) |
|setup (roxie) |19 |19 |0 |18 sec  (00:00:18) |
|test (hthor) |983 |983 |0 |821 sec  (00:13:41) |
|test (thor) |893 |893 |0 |986 sec  (00:16:26) |
|test (roxie) |1052 |1052 |0 |865 sec  (00:14:25) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   22 sec (00:00:22) |  437 sec (00:07:17) |  103 sec (00:01:43) |   20 sec (00:00:20) |   16 sec (00:00:16) | 1241 sec (00:20:41) |   18 sec (00:00:18) | 1857 sec (00:30:57) |

HPCCSmoketest commented:
Process of PR-15626, label: issue26916 starts now.
The reason of this test is: Forced to re-test.
Commit ID: 68423deaa4bd32d4603a3c4af3fec514ccf91eec
Estimated completion time is 0.75 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 Host: ip-10-20-0-226.ca-central-1.compute.internal
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Git: 2.9.5,  CMake: 3.22.1,  cUrl: 7.67.0,  node.js: v16.13.1,  npm: 8.1.2

richardkchapman commented on common/remote/hooks/git/gitfile.cpp:
```diff
@@ -546,26 +324,27 @@ class NewGitRepositoryFileIO : implements CSimpleInterfaceOf<IFileIO>
 class NewGitRepositoryFile : public GitRepositoryFile
```
What's the purpose of having the base class now?

ghalliday replied:
none really.  I'll combine, and rename (to remove the New prefixes)

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 Host: ip-10-20-0-226.ca-central-1.compute.internal
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Git: 2.9.5,  CMake: 3.22.1,  cUrl: 7.67.0,  node.js: v16.13.1,  npm: 8.1.2
Sha: 68423deaa4bd32d4603a3c4af3fec514ccf91eec
Containerized:False
Build: success
Milestone:Install hpccsystems-platform-community_8.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |222 |222 |0 |0 |0 |64 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |9 |9 |0 |21 sec  (00:00:21) |
|setup (thor) |9 |9 |0 |38 sec  (00:00:38) |
|setup (roxie) |19 |19 |0 |18 sec  (00:00:18) |
|test (hthor) |983 |983 |0 |818 sec  (00:13:38) |
|test (thor) |893 |893 |0 |987 sec  (00:16:27) |
|test (roxie) |1052 |1052 |0 |867 sec  (00:14:27) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   27 sec (00:00:27) |  433 sec (00:07:13) |  101 sec (00:01:41) |   21 sec (00:00:21) |   16 sec (00:00:16) | 1240 sec (00:20:40) |   17 sec (00:00:17) | 1855 sec (00:30:55) |

HPCCSmoketest commented:
Process of PR-15626, label: issue26916 starts now.
The reason of this test is: Forced to re-test.
Commit ID: 341aee43369f37ac1202cab01c4cb965df769e75
Estimated completion time is 0.75 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 Host: ip-10-20-0-39.ca-central-1.compute.internal
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Git: 2.9.5,  CMake: 3.22.1,  cUrl: 7.67.0,  node.js: v16.13.1,  npm: 8.1.2

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 Host: ip-10-20-0-39.ca-central-1.compute.internal
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Git: 2.9.5,  CMake: 3.22.1,  cUrl: 7.67.0,  node.js: v16.13.1,  npm: 8.1.2
Sha: 341aee43369f37ac1202cab01c4cb965df769e75
Containerized:False
Build: success
Milestone:Install hpccsystems-platform-community_8.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |222 |222 |0 |0 |0 |63 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |9 |9 |0 |21 sec  (00:00:21) |
|setup (thor) |9 |9 |0 |38 sec  (00:00:38) |
|setup (roxie) |19 |19 |0 |17 sec  (00:00:17) |
|test (hthor) |983 |983 |0 |805 sec  (00:13:25) |
|test (thor) |893 |893 |0 |972 sec  (00:16:12) |
|test (roxie) |1052 |1052 |0 |911 sec  (00:15:11) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   27 sec (00:00:27) |  431 sec (00:07:11) |  100 sec (00:01:40) |   19 sec (00:00:19) |   18 sec (00:00:18) | 1225 sec (00:20:25) |   18 sec (00:00:18) | 1838 sec (00:30:38) |

richardkchapman reviewed (approved):
Looks much cleaner now. Please squash

HPCCSmoketest commented:
Process of PR-15626, label: issue26916 starts now.
The reason of this test is: Forced to re-test.
Commit ID: a38ba9cac83e2d3c14f3db8ef17084a6d9a3d9c1
Estimated completion time is 0.75 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 Host: ip-10-20-0-174.ca-central-1.compute.internal
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Git: 2.9.5,  CMake: 3.22.1,  cUrl: 7.67.0,  node.js: v16.13.1,  npm: 8.1.2

ghalliday commented:
@richardkchapman I have squashed (and added some default initialisers onto some class members to clean it up rather than fix an issue).
I think this would be suitable for older branches, but have a reservation.  Is the libgit2 being installed correctly?  Initially it was building but not installing.  I think that is now fixed, but there are no regression tests so not really tested.  Can you think of any good ways to add a regression test?  Should we extend the regression suite to allow links that compile from a url?  Should I add a weird ecl format @..... or <link>...</link> to implement this in the compiler?

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 Host: ip-10-20-0-174.ca-central-1.compute.internal
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Git: 2.9.5,  CMake: 3.22.1,  cUrl: 7.67.0,  node.js: v16.13.1,  npm: 8.1.2
Sha: a38ba9cac83e2d3c14f3db8ef17084a6d9a3d9c1
Containerized:False
Build: success
Milestone:Install hpccsystems-platform-community_8.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |222 |222 |0 |0 |0 |64 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |9 |9 |0 |21 sec  (00:00:21) |
|setup (thor) |9 |9 |0 |38 sec  (00:00:38) |
|setup (roxie) |19 |19 |0 |17 sec  (00:00:17) |
|test (hthor) |983 |983 |0 |825 sec  (00:13:45) |
|test (thor) |893 |893 |0 |991 sec  (00:16:31) |
|test (roxie) |1052 |1052 |0 |871 sec  (00:14:31) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   23 sec (00:00:23) |  431 sec (00:07:11) |  102 sec (00:01:42) |   19 sec (00:00:19) |   17 sec (00:00:17) | 1245 sec (00:20:45) |   18 sec (00:00:18) | 1855 sec (00:30:55) |

HPCCSmoketest commented:
Process of PR-15626, label: issue26916 starts now.
The reason of this test is: Forced to re-test.
Commit ID: f17fb991ff8be3173d444473e482ba7d1a5a49cf
Estimated completion time is 0.75 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 Host: ip-10-20-0-172.ca-central-1.compute.internal
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Git: 2.9.5,  CMake: 3.22.1,  cUrl: 7.67.0,  node.js: v16.13.1,  npm: 8.1.2

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 Host: ip-10-20-0-172.ca-central-1.compute.internal
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Git: 2.9.5,  CMake: 3.22.1,  cUrl: 7.67.0,  node.js: v16.13.1,  npm: 8.1.2
Sha: f17fb991ff8be3173d444473e482ba7d1a5a49cf
Containerized:False
Build: success
Milestone:Install hpccsystems-platform-community_8.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |222 |222 |0 |0 |0 |64 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |0 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |9 |9 |0 |21 sec  (00:00:21) |
|setup (thor) |9 |9 |0 |39 sec  (00:00:39) |
|setup (roxie) |19 |19 |0 |18 sec  (00:00:18) |
|test (hthor) |983 |983 |0 |826 sec  (00:13:46) |
|test (thor) |893 |893 |0 |984 sec  (00:16:24) |
|test (roxie) |1052 |1052 |0 |923 sec  (00:15:23) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   22 sec (00:00:22) |  435 sec (00:07:15) |  104 sec (00:01:44) |   23 sec (00:00:23) |   16 sec (00:00:16) | 1239 sec (00:20:39) |   18 sec (00:00:18) | 1857 sec (00:30:57) |
